<?xml version="1.0"?>
<launch>
  <arg name="debug"  default="False" />
  <arg name="bag_name_"  			default="recorder_save.bag" />
  <arg name="folder_name_"  	default="/kinect_record_saves/" />
  <arg name="single_frame_"  	default="False" />
  <arg name="viewer_off_" 		default="False" />
  <arg name="delay_time_"			default="0.02" />
  
  <arg name="blur_param_"			default="5" />
  <arg name="hsv_target_"			default="145" />
  <arg name="hsv_threshold_"	default="10" />
  <arg name="contour_area_min_"	default="150" />
  <arg name="contour_area_max_"	default="500" />
  <arg name="contour_ratio_min_"	default="6" />
  <arg name="contour_ratio_max_"	default="9" />
  
  <node name="rosbag_replay" pkg="ivan_aruco" type="rosbag_replay" output="screen"> 
    <param name="debug"           	type="bool"   	value="$(arg debug)"/> 
    <param name="bag_name_"      		type="string"   value="$(arg bag_name_)"/> 
    <param name="folder_name_"     	type="string"   value="$(arg folder_name_)"/> 
    <param name="single_frame_"    	type="bool"   	value="$(arg single_frame_)"/> 
    <param name="viewer_off_"    		type="bool"   	value="$(arg viewer_off_)"/> 
    <param name="delay_time_"				type="double"		value="$(arg delay_time_)"/>
   </node>
  
  <node name="replay_analysis" pkg="ivan_aruco" type="replay_analysis" output="screen"> 
		<param name="blur_param_"          	type="int"   	value="$(arg blur_param_)"/> 
		<param name="hsv_target_"          	type="int"   	value="$(arg hsv_target_)"/> 
		<param name="hsv_threshold_"   			type="int"   	value="$(arg hsv_threshold_)"/> 
		<param name="contour_area_min_"    	type="int"   	value="$(arg contour_area_min_)"/> 
		<param name="contour_area_max_"     type="int"   	value="$(arg contour_area_max_)"/> 
		<param name="contour_ratio_min_"    type="double"   	value="$(arg contour_ratio_min_)"/> 
		<param name="contour_ratio_max_"    type="double"   	value="$(arg contour_ratio_max_)"/> 
	</node>
	
</launch>


